File: KevinRene/src/qls/ast/Identifier.java
package qls.ast;

import ql.ast.expression.QLType;
import ql.ast.expression.type.QLError;
import qls.ast.visitor.Visitor;

public class Identifier implements QLSNode {
	private final String identifier;
	
	public Identifier(String identifier) {
		this.identifier = identifier;
	}
	
	public QLType getType() {
		return new QLError();
	}
	
	@Override
	public <T> T accept(Visitor<T> visitor) {		
		return visitor.visit(this);
	}

	@Override
	public String toString() {
		return identifier;
	}

}

File: CarlaErik/UvA.SoftCon.Questionnaire/QLS/AST/Model/Identifier.cs
using UvA.SoftCon.Questionnaire.Common.AST.Model;

namespace UvA.SoftCon.Questionnaire.QLS.AST.Model
{
    /// <summary>
    /// A name that refers to a question in the QL AST.
    /// </summary>
    public class Identifier : QLSNode
    {
        public string Name
        {
            get;
            private set;
        }

        internal Identifier(string name, TextPosition position)
            : base(position)
        {
            Name = name;
        }

        public override T Accept<T>(IStyleSheetVisitor<T> visitor)
        {
            return visitor.VisitIdentifier(this);
        }

        public override string ToString()
        {
            return Name;
        }
    }
}
